fastpair/rust/src/bluetooth/unsupported/adapter.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use async_trait::async_trait;
+
+use super::BleDevice;
+use crate::bluetooth::common::Adapter;
+
+/// Concrete type implementing `Adapter`, used for unsupported devices.
+/// Every method should panic.
+pub struct BleAdapter;
+
+#[async_trait]
+impl Adapter for BleAdapter {
+    type Device = BleDevice;
+
+    async fn default() -> Result<Self, anyhow::Error> {
+        panic!("Unsupported target platform.");
```

Comment:
there is also `unimplemented!` which does the same thing, and maybe is more communicative to the reader? Or not. I'm not sure which is better.

Reply:
I personally prefer `panic!` here. If there's a specific platform with pending features then `unimplemented!` makes more sense, but for an entire unsupported platform what's unimplemented isn't the specific function, but rather the entire platform library.